pkg/generate/app/cmd/newapp.go

```diff
@@ -710,6 +711,23 @@ func (c *AppConfig) Run() (*AppResult, error) {
 		}
 	}
 
+	// before returning, if this is a template run, and environment variables were passed in, let's apply to the environment variables
+	// to every pod template object
+	if len(templateObjects) > 0 {
+		for i := range objects {
```

Comment:
Yeah, in the spirit of this, I moved (and will soon push) the loop into the `buildTemplate` method.  It seems to work correctly based on local testing, but let me know if I'm missing some ramification of such a move.